packages/collector/test/tracing/cloud/azure/blob/test.js

```diff
@@ -0,0 +1,477 @@
+const { verifyHttpRootEntry, expectExactlyNMatching } = require('../../../../../../core/test/test_util');
+const ProcessControls = require('../../../../test_util/ProcessControls');
+const globalAgent = require('../../../../globalAgent');
+const { fail } = expect;
+const testUtils = require('../../../../../../core/test/test_util');
+const { createContainer, deleteContainer, minimumNodeJsVer } = require('./util');
+const { BlobServiceClient } = require('@azure/storage-blob');
+const containerName = `nodejs-team-${uuid()}`;
+const storageAccount = process.env.AZ_STORAGE_ACCOUNT_NAME;
+const accountKey = process.env.AZ_STORAGE_ACCOUNT_KEY;
+
+/**
+ * endPoint is assigned with the default constant string for appending EndpointSuffix
+ * to connection string of an azure account
+ */
+const endPoint = 'EndpointSuffix=core.windows.net';
+const connStr = `DefaultEndpointsProtocol=https;AccountName=${storageAccount};AccountKey=${accountKey};${endPoint}`;
+const blobServiceClient = BlobServiceClient.fromConnectionString(connStr);
+const containerClient = blobServiceClient.getContainerClient(containerName);
+
+let mochaSuiteFn = supportedVersion(process.versions.node) ? describe : describe.skip;
+mochaSuiteFn = semver.lt(process.versions.node, minimumNodeJsVer) ? describe.skip : mochaSuiteFn;
+
+/**
+ * This suite is skipped if no storageAccount or accountKey has been provided via AZ_STORAGE_ACCOUNT_NAME
+ * and AZ_STORAGE_ACCOUNT_KEY. For the Azure blob tests, the azure storage account used is teamnodejstracer
+ * which is the value for AZ_STORAGE_ACCOUNT_NAME. From the azure portal, navigate to this storage account
+ * and under the Access keys, Key can be found for AZ_STORAGE_ACCOUNT_KEY.
+ */
+if (!storageAccount || !accountKey) {
+  describe('tracing/cloud/azure/blob', function () {
+    it('The configuration for Azure is missing', () => {
+      fail('Please set process.env.AZ_STORAGE_ACCOUNT_KEY and process.env.AZ_STORAGE_ACCOUNT_NAME before tests.');
+    });
+  });
+} else {
+  mochaSuiteFn('tracing/cloud/azure/blob', function () {
+    globalAgent.setUpCleanUpHooks();
+    const agentControls = globalAgent.instance;
+
+    const controls = new ProcessControls({
+      dirname: __dirname,
+      useGlobalAgent: true,
+      env: {
+        AZURE_CONTAINER_NAME: containerName,
+        AZURE_CONNECTION_STRING: connStr,
+        AZURE_STORAGE_ACCOUNT: storageAccount,
+        AZURE_ACCOUNT_KEY: accountKey
+      }
+    });
+    ProcessControls.setUpHooks(controls);
+    this.timeout(config.getTestTimeout());
+
+    before(async () => {
+      await createContainer(containerClient);
+    });
+
+    after(async () => {
+      await deleteContainer(containerClient);
+    });
+
+    it('uploads block data', async () => {
+      await controls.sendRequest({
+        method: 'GET',
+        path: '/uploadDataBlock'
+      });
+      await testUtils.retry(async () => {
+        const spans = await agentControls.getSpans();
```

Comment:
Please check the number of spans, see [my comment.](https://github.com/instana/nodejs/pull/967#discussion_r1430014856)

Reply:
> opSpanCount: 1

Please add the number of *all* spans. 
e.g. 2 http calls + azure call + fs call = 4

We always expect one azure span, please use "expectExactlyOneMatching".

Reply:
resolved